src/newtoncg.jl

```diff
@@ -22,60 +51,59 @@ function linesearch_args(
 end
 
 mutable struct KL_
```

Comment:
Can you replace both `KL_` and `∇KL_` with a single `EvalCount`? Make make the struct immutable and use an `Base.Threads.Atomic{Int}` as counter.